packages/core/src/modules/proofs/services/ProofService.ts

```diff
@@ -1037,6 +1046,16 @@ export class ProofService {
 
     return credentialDefinitions
   }
+
+  public checkAttributePredicateMatch(proofRequest: ProofRequest) {
+    for (const attribute of proofRequest.requestedAttributes.keys()) {
+      for (const predicate of proofRequest.requestedPredicates.keys()) {
+        if (attribute === predicate) {
+          throw new AriesFrameworkError(`The proof request contains attributes that match the predicates: ${attribute}`)
+        }
+      }
+    }
+  }
```

Comment:
For each key it will loop through the keys of the other array. Maybe we can use `.includes` here?

```suggestion
    const predicateGroupNames = proofRequest.requestedPredicates.keys()
    for (const attribute of proofRequest.requestedAttributes.keys()) {
      if (predicateGroupNames.includes(attribute)) {
        throw new AriesFrameworkError(`The proof request contains attributes that match the predicates:  ${attribute}`)
        }
      }
  }
```